src/apply_imputation_link.py

```diff
@@ -0,0 +1,161 @@
+def create_and_merge_imputation_values(
```

Comment:
I believe this function imputes the values by creating new columns and merging them, so create and merge should go in the docstrings.

I understand this might crossover with the name of higher level function perhaps rename it to add_imputed_values?